### ISSUE TITLE
matplotlib >= 3.8 breaks plots

### ISSUE THREAD

**Issue by bmaranville** [bumps/bumps#129]
The code in bumps.plotutil.next_color breaks for the newest matplotlib.  

This worked in matplotlib 3.7.1:
```python
    if axes is None:
        axes = pylab.gca()
    lines = axes._get_lines
    try:
        base = next(lines.prop_cycler)['color']
```

but in 3.8 lines has no attribute `prop_cycler`

This diff seems to fix the problem, but I don't know if it does what is intended:
```diff
diff --git a/bumps/plotutil.py b/bumps/plotutil.py
index e719e64..14fa2d0 100644
--- a/bumps/plotutil.py
+++ b/bumps/plotutil.py
@@ -56,12 +56,15 @@ def next_color(axes=None):
         axes = pylab.gca()
     lines = axes._get_lines
     try:
-        base = next(lines.prop_cycler)['color']
+        base = lines.get_next_color()
     except Exception:
-        try: # Cruft 1.4-1.6?
-            base = next(lines.color_cycle)
-        except Exception:  # Cruft 1.3 and earlier
-            base = lines._get_next_cycle_color()
+        try: # Cruft 1.7 - 3.7?
+            base = next(lines.prop_cycler)['color']
+        except Exception:
+            try: # Cruft 1.4-1.6?
+                base = next(lines.color_cycle)
+            except Exception:  # Cruft 1.3 and earlier
+                base = lines._get_next_cycle_color()
     return base
 
 
```

**Comment by bmaranville** [comment]
I was trying to figure out if there is a way to replace this use of a private API (axes._get_lines) with something like

prop_cycle = pylab.rcParams['axes.prop_cycle']

but matplotlib Cycler objects are iterable but not iterators, so I don't know how to maintain the "hidden state" implied by the `next_color()` command.

In matplotlib <= 3.7 gca()._get_lines.prop_cycler returned an instance of `itertools.cycle`, which was an iterator, so you could call "next" on it.  Now I'm not sure what the best way to proceed is.

**Comment by bmaranville** [comment]
resolved by ddfae0d
